monitoring/monitoring.go

```diff
@@ -43,25 +37,42 @@ type (
 		// this is equivalent to subtracting -1 to the current value. For Gauges,
 		// this is equivalent to setting the value to -1. For Distributions,
 		// this is equivalent to making an observation of value -1.
-		Decrement()
+		//
+		// Not used in istio, removed pending new interface.
+		//Decrement()
 
 		// Name returns the name value of a Metric.
+		// TODO: internal use only
 		Name() string
 
 		// Record makes an observation of the provided value for the given measure.
+		// Majority of Istio is setting this to an int
 		Record(value float64)
 
 		// RecordInt makes an observation of the provided value for the measure.
-		RecordInt(value int64)
+		// Not actually used in Istio.
+		//RecordInt(value int64)
+
+		// Prometheus uses ConterVec and With(map[string]string) returning Counter
+		//
+		// Expvar doesn't support labels - but can be used in the name, creating a new expvar in a map
+		// The labels are really made part of the name when exporting and is the name of the TS
+		//
+		// Otel uses attribute.Int("name", val) and similar - from the stable package -
+		// but doesn't create a new Metric, it is an option to Add().
```

Comment:
Not pre-creating objects with the attributes is what was costing Istiod to spend ~25% of its time in metrics code. So hopefully this is not suggesting we do that?